core/arch/instrlist.c

```diff
@@ -245,7 +245,8 @@ check_translation(instrlist_t *ilist, instr_t *inst)
 #if defined(CLIENT_INTERFACE) && defined(ARM)
     if (instr_is_meta(inst)) {
         dr_pred_type_t auto_pred = ilist->auto_pred;
-        if (auto_pred != DR_PRED_AL && auto_pred != DR_PRED_NONE) {
+        if (auto_pred != DR_PRED_AL && auto_pred != DR_PRED_OP &&
```

Comment:
Ack. 

By the way, I misunderstood the usage of `instr_predicate_is_cond()`. I thought this was to be used in place of the `DR_PRED_OP` check only, but it seems to replace all three checks for `DR_PRED_NONE`, `DR_PRED_AL` and `DR_PRED_OP`. Will update the usage of this method accordingly in my CL.